### PR TITLE
[WIP] Add bindings for Karma proto

### LINKED ISSUE #197: Add bindings for Karma contract
Need to be able to deposit/withdraw coins to/from the Karma contract.

### PR DESCRIPTION
**Pull request by gakonst**
Proto bindings from https://github.com/loomnetwork/go-loom/pull/190

Karma contract bindings from https://github.com/loomnetwork/loomchain/pull/466

Closes #197

### CODE REVIEW

File: src/contracts/karma.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { Contract } from '../contract'
+import { Address } from '../address'
+import {
+  KarmaAddressSource, KarmaUserAmount, KarmaStateKeyUser, KarmaSources, KarmaSourceReward, KarmaState, KarmaSourceTarget, KarmaUserTarget, KarmaTotal
+} from '../proto/karma_pb'
+import { unmarshalBigUIntPB, marshalBigUIntPB } from '../big-uint'
+
+export class Karma extends Contract {
+  static async createAsync(client: Client, callerAddr: Address): Promise<DPOS> {
+    const contractAddr = await client.getContractAddressAsync('dpos')
```
Comment (enlight) [inline src/contracts/karma.ts]: lol, copy/pasta

File: src/contracts/karma.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    return new Karma({ contractAddr, callerAddr, client })
+  }
+
+  constructor(params: { contractAddr: Address; callerAddr: Address; client: Client }) {
+    super(params)
+  }
+
+  async depositCoinAsync(amount: BN, from: Address): Promise<void> {
+    const karmaUserAmountReq = new KarmaUserAmount()
```
Comment (enlight) [inline src/contracts/karma.ts]: `KarmaUserAmount` gah. nvm. I'll fix it later.

File: src/contracts/karma.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      'GetSources',
+      user.MarshalPB(),
+      response
+    )
+    return result
+  }
+
+  async getUserKarmaAsync(user: Address, target: KarmaSourceTarget): Promise<KarmaTotal> {
+    const karmaUserTargetReq = new KarmaUserTarget()
+    let response: any
```
Comment (enlight) [inline src/contracts/karma.ts]: This should be a protobuf instance?

File: src/contracts/karma.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const result = await this.staticCallAsync(
+      'GetUserState',
+      user.MarshalPB(),
+      response
+    )
+    return result
+  }
+
+  async getSourcesAsync(user: Address): Promise<KarmaSources> {
+    let response: any
```
Comment (enlight) [inline src/contracts/karma.ts]: This should be a protobuf instance?

File: src/contracts/karma.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  async withdrawCoinAsync(amount: BN, to: Address): Promise<void> {
+    const karmaUserAmountReq = new KarmaUserAmount()
+    karmaUserAmountReq.setAmount(marshalBigUIntPB(amount))
+    karmaUserAmountReq.setUser(to.MarshalPB())
+    return this.callAsync<void>('WithdrawCoin', karmaUserAmountReq)
+  }
+
+  async getUserStateAsync(user: Address): Promise<KarmaState> {
+    let response: any
```
Comment (enlight) [inline src/contracts/karma.ts]: This should be a protobuf instance?

File: src/contracts/karma.ts (hunk 3 above)
Comment (gakonst) [reply src/contracts/karma.ts]: Yeah, I wanna add a new proto response and wrap it inside, as we do in other places.

File: src/contracts/karma.ts (hunk 1 above)
Comment (gakonst) [reply src/contracts/karma.ts]: Busted lol

**Pull request closed without merging**